lib/pubmed_client.rb

```diff
@@ -15,25 +15,25 @@ def fetch_records_for_pmid_list(pmids)
     pmidValuesForPost = pmid_list.collect { |pmid| "&id=#{pmid}" }.join
     timeout_retries ||= 3
     timeout_period ||= 500
-    http = Net::HTTP.new(Settings.PUBMED.HOST)
-    http.read_timeout = timeout_period
-    request = Net::HTTP::Post.new(Settings.PUBMED.FETCH_PATH)
-    request.body = pmidValuesForPost
-    # http.start
-    the_incoming_xml = http.request(request).body
-    # http.finish
-    the_incoming_xml
 
-    rescue Timeout::Error => te
-      timeout_retries -= 1
-      if timeout_retries > 0
-        # increase timeout
-        timeout_period = + 500
-        retry
-      else
-        NotificationManager.error(te, 'Timeout::Error during PubMed Fetch API call', self)
-        raise
-      end
+    conn = Faraday.new(url: "https://#{Settings.PUBMED.HOST}") do |faraday|
+      faraday.request :retry, max: timeout_retries,
+        interval: 0.5,
+        interval_randomness: 0.5,
+        backoff_factor: 2
+      faraday.adapter :httpclient
+    end
+    conn.options.timeout = timeout_period
+    conn.options.open_timeout = 10
+
+    response = conn.send(:post) do |req|
+      req.url Settings.PUBMED.FETCH_PATH
+      req.headers['Host'] = Settings.PUBMED.HOST
+      req.body = pmidValuesForPost
+    end
+
+    response.body
+
```

Comment:
There are tests here that use cassettes to verify the connection: https://github.com/sul-dlss/sul_pub/blob/master/spec/lib/pubmed_client_spec.rb

Do you think we need something else?  And stubbing the call out without actually making it?